tensorbay/client/dataset.py

```diff
@@ -585,10 +622,40 @@ def copy_segment(
                 2. "override": the source segment will override the origin segment;
                 3. "skip": keep the origin segment.
 
-        Returns:  #noqa: DAR202
+        Returns:
             The client of the copied target segment.
 
+        Raises:
+            InvalidParamsError: When the strategy is invalid.
+            OperationError: When copying the segment to the same location.
+
         """
+        if not source_client and not target_name:
```

Comment:
it is better to add a new function to the base class to avoid the redundant logic